pkg/distro/rhel/rhel9/azure.go

```diff
@@ -582,3 +637,94 @@ func defaultAzureImageConfig(rd *rhel.Distribution) *distro.ImageConfig {
 func sapAzureImageConfig(rd *rhel.Distribution) *distro.ImageConfig {
 	return sapImageConfig(rd.OsVersion()).InheritFrom(defaultAzureImageConfig(rd))
 }
+
+func azureCVMImageConfig(rd *rhel.Distribution) *distro.ImageConfig {
+	ic := &distro.ImageConfig{
+		Timezone: common.ToPtr("Etc/UTC"),
+		Locale:   common.ToPtr("en_US.UTF-8"),
+		Keyboard: &osbuild.KeymapStageOptions{
+			Keymap: "us",
+			X11Keymap: &osbuild.X11KeymapOptions{
+				Layouts: []string{"us"},
+			},
+		},
+		EnabledServices: []string{
+			"cloud-config",
+			"cloud-final",
+			"cloud-init",
+			"cloud-init-local",
+			"NetworkManager",
+			"nm-cloud-setup.service",
+			"nm-cloud-setup.timer",
+			"sshd",
+			"waagent",
+		},
+		MaskedServices: []string{
+			"firewalld.service",
+		},
+
+		DefaultTarget: common.ToPtr("multi-user.target"),
+		Sysconfig: []*osbuild.SysconfigStageOptions{
+			{
+				NetworkScripts: &osbuild.NetworkScriptsOptions{
+					IfcfgFiles: map[string]osbuild.IfcfgFile{
+						"eth0": {
+							Device:    "eth0",
+							Bootproto: osbuild.IfcfgBootprotoDHCP,
+						},
+					},
+				},
+			},
+		},
+
+		DefaultKernelName: common.ToPtr("kernel-uki-virt"),
+		NoBLS:             common.ToPtr(true),
+	}
+
+	if rd.IsRHEL() {
+		// NOTE: copied from the rest of the Azure image types
+		ic.GPGKeyFiles = append(ic.GPGKeyFiles, "/etc/pki/rpm-gpg/RPM-GPG-KEY-redhat-release")
+	}
+
+	return ic
+}
+
+func azureCVMPartitionTables(t *rhel.ImageType) (disk.PartitionTable, bool) {
+	switch t.Arch().Name() {
+	case arch.ARCH_X86_64.String():
+		return disk.PartitionTable{
+			UUID: "D209C89E-EA5E-4FBD-B161-B461CCE297E0",
+			Type: disk.PT_GPT,
+			Partitions: []disk.Partition{
+				{
+					Size: 252 * datasizes.MebiByte,
+					Type: disk.EFISystemPartitionGUID,
+					UUID: disk.EFISystemPartitionUUID,
+					Payload: &disk.Filesystem{
+						Type:         "vfat",
+						UUID:         disk.EFIFilesystemUUID,
+						Mountpoint:   "/boot/efi",
+						Label:        "EFI-SYSTEM",
+						FSTabOptions: "defaults,uid=0,gid=0,umask=077,shortname=winnt",
+						FSTabFreq:    0,
+						FSTabPassNo:  2,
+					},
+				},
+				{
+					Size: 5 * datasizes.GibiByte,
+					Type: disk.RootPartitionX86_64GUID,
+					Payload: &disk.Filesystem{
+						Type:         "xfs",
```

Comment:
Shouldn't this be `ext4`? I think I saw it in the kickstarts that Vitaly gave us. I think it must be ext4 because the deployment script is shrinking this filesystem, so it can be put into a LUKS container with the same size (so the partition_size = luks_header + filesystem).

Reply:
Right!  Thanks for catching it.